src/common/com/intellij/plugins/haxe/model/HaxePackageModel.java

```diff
@@ -125,9 +125,23 @@ public HaxeFileModel getFileModel(String fileName) {
   protected HaxeFile getFile(String fileName) {
     PsiDirectory directory = root.access(path);
     if (directory != null && directory.isValid()) {
-      PsiFile file = directory.findFile(fileName + ".hx");
-      if (file != null && file.isValid() && file instanceof HaxeFile) {
-        return (HaxeFile)file;
+      PsiFile file;
+      try {
+        String fName = fileName + ".hx";
+        //TODO @ Eric.  What could be the source of a request for "Unknown.hx" ?
```

Comment:
There is a magic type of "unknown" in the language that is assigned to variables before type inference is able to set the type.  (It's the same in the compiler -- just `trace(some_uninitialized_var);` to see it pop out.)  If the resolver gets a variable in this condition, it's going to try and find the class by upper-casing the name and looking for an implementation file.  Were we throwing an exception out of this function?  What was the bug that made you look here?  

(BTW, ProcessCanceledExceptions are expected and *should* cancel the current task.  That's the way that IntelliJ deals with blocked/blocking tasks.)

Reply:
Yeah whenever it tried to load "Unknown.hx" it was throwing.  The Intellij stack trace got me there eventually.

Reply:
Oh I just noticed your BTW.  That may be true but when it threw here it stopped parsing code.  This is evident if you can repro from my instructions in the commit notes.

Reply:
I've been unable to repro the stop exception being thrown.  If I set a breakpoint in `getFile()`, I can definitely see the code trying to find the file -- every time an unknown class reference is created.  Issue #938 is for that.